include/SFML/System/Exception.hpp

```diff
@@ -0,0 +1,54 @@
+////////////////////////////////////////////////////////////
+//
+// SFML - Simple and Fast Multimedia Library
+// Copyright (C) 2007-2024 Laurent Gomila (laurent@sfml-dev.org)
+//
+// This software is provided 'as-is', without any express or implied warranty.
+// In no event will the authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose,
+// including commercial applications, and to alter it and redistribute it freely,
+// subject to the following restrictions:
+//
+// 1. The origin of this software must not be misrepresented;
+//    you must not claim that you wrote the original software.
+//    If you use this software in a product, an acknowledgment
+//    in the product documentation would be appreciated but is not required.
+//
+// 2. Altered source versions must be plainly marked as such,
+//    and must not be misrepresented as being the original software.
+//
+// 3. This notice may not be removed or altered from any source distribution.
+//
+////////////////////////////////////////////////////////////
+
+#pragma once
+
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/System/Export.hpp>
+
+#include <stdexcept>
+
+
+namespace sf
+{
+////////////////////////////////////////////////////////////
+/// \brief Library-specific exception type
+///
+////////////////////////////////////////////////////////////
+class SFML_SYSTEM_API Exception : public std::runtime_error
+{
+public:
+    ////////////////////////////////////////////////////////////
+    /// \brief Construct with message
+    ///
+    /// \param message Exception message
+    ///
+    ////////////////////////////////////////////////////////////
+    explicit Exception(const std::string& message) : std::runtime_error(message)
+    {
+    }
```

Comment:
Should we keep the impl in the header?

Reply:
I would say if there is not a reason to make it in the header then why not do it in the implementation file and imrpove compile times and be consistent